wzdx/raw_to_standard/icone.py

```diff
@@ -0,0 +1,223 @@
+import argparse
+import json
+import logging
+import time
+import uuid
+import xml.etree.ElementTree as ET
+from collections import OrderedDict
+
+from wzdx.tools import date_tools, polygon_tools, wzdx_translator
+from wzdx.util.collections import PathDict
+
+PROGRAM_NAME = 'iConeRawToStandard'
+PROGRAM_VERSION = '1.0'
+
+
+def main():
+    input_file, output_dir = parse_rtdh_arguments()
+    input_file_contents = open(input_file, 'r').read()
+    generated_messages = generate_standard_messages_from_string(
+        input_file_contents)
+
+    generated_files_list = []
+    for message in generated_messages:
+        output_path = f"{output_dir}/standard_icone_{message['event']['source']['id']}_{round(message['rtdh_timestamp'])}.json"
+        open(output_path, 'w+').write(json.dumps(message, indent=2))
+        generated_files_list.append(output_path)
+
+    if generated_files_list:
+        print(
+            f"Successfully generated standard message files: {generated_files_list}")
+    else:
+        logging.warning(
+            "Standard message generation failed. See messages printed above")
+
+
+def generate_standard_messages_from_string(input_file_contents):
+    raw_messages = generate_raw_messages(input_file_contents)
+    standard_messages = []
+    for message in raw_messages:
+        standard_messages.append(
+            generate_rtdh_standard_message_from_raw_single(message))
+    return standard_messages
+
+
+def generate_raw_messages(message, invalid_messages_callback=None):
+    response_xml = ET.fromstring(message)
+    msg_lst = response_xml.findall('incident')
+    messages = []
+
+    # Loop through all elements and print each element to PubSub
+    for msg in msg_lst:
+        print(msg)
+        message = ET.tostring(msg, encoding='utf8')
+        obj = wzdx_translator.parse_xml_to_dict(message)
+        if not validate_incident(obj.get('incident', {})):
+            if invalid_messages_callback:
```

Comment:
Here we're re-scoping a "message" variable. Should we change the passed in variable to be `input_file_contents` to match the other methods in this file and clear up confusion?

Reply:
updated to be incident